File: diploma-rest-api/src/main/java/org/arhor/diploma/exception/EntityNotFoundException.java
package org.arhor.diploma.exception;

public class EntityNotFoundException extends RuntimeException {

  private final String className;
  private final String fieldName;
  private final Object fieldValue;

  public EntityNotFoundException(String className, String fieldName, Object fieldValue) {
    this.className = className;
    this.fieldName = fieldName;
    this.fieldValue = fieldValue;
  }

  public String getClassName() {
    return className;
  }

  public String getFieldName() {
    return fieldName;
  }

  public Object getFieldValue() {
    return fieldValue;
  }
}

File: diploma-server/src/main/kotlin/org/arhor/diploma/exception/EntityNotFoundException.kt
package org.arhor.diploma.exception

private const val UNKNOWN = "unknown"

private val String?.safe: String
    get() = this ?: UNKNOWN

data class EntityNotFoundException(
    val entityType: String?,
    val propName: String?,
    val propValue: Any?
) : RuntimeException("[${entityType.safe}] with property [${propName.safe}]=[${propValue}] is not found")

